src/NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp

```diff
@@ -98,55 +102,41 @@ auto partial_derivatives(
                                   tmpl::size_t<Dim>, DerivativeFrame>>;
 
 namespace Tags {
-namespace Tags_detail {
-template <typename Tag, typename VolumeDim, typename Frame>
-struct deriv_impl<Tag, VolumeDim, Frame,
-                  Requires<tt::is_a_v<Tensor, db::item_type<Tag>>>>
-    : db::PrefixTag, db::SimpleTag {
-  using type = TensorMetafunctions::prepend_spatial_index<
-      db::item_type<Tag>, VolumeDim::value, UpLo::Lo, Frame>;
-  using tag = Tag;
-  static std::string name() noexcept { return "deriv"; }
-};
 
-// Partial derivatives with derivative index in the frame related to the logical
-// coordinates by InverseJacobianTag
-template <typename... VariablesTags, typename... DerivativeTags,
-          typename InverseJacobianTag>
-struct deriv_impl<
-    tmpl::list<VariablesTags...>, tmpl::list<DerivativeTags...>,
-    InverseJacobianTag,
-    Requires<tt::is_a_v<Tensor, db::item_type<InverseJacobianTag>>>>
-    : db::ComputeTag {
+/*!
+ * \ingroup DataBoxTagsGroup
+ * \brief Compute the spatial derivatives of tags in a Variables
+ *
+ * Computes the spatial derivatives of the Tensors in the Variables represented
+ * by `VariablesTag` in the frame mapped to by `InverseJacobianTag`. You can
+ * optionally specify a subset of the tags in the `Variables` as `GradientTags`
+ * to only take the gradient of those.
+ *
+ * This tag inherits from `db::add_prefix_tag<Tags::deriv, VariablesTag>`.
```

Comment:
I don't think this inheritance is right.  Add a test that you can retrieve the computed Variables through the simple tag.

Reply:
Yeah something is wrong. Do you know how I can make the tag derive from the  `db::add_prefix_tag<Tags::deriv, VariablesTag>` with a `tags_list` that is restricted to the `GradientTags`?

Reply:
OK, yeah, that's a lot trickier than I thought it would be.  I'll submit a PR with a metafunction to set the tags_list of a Variables tag preserving prefixes shortly.  I think that along with things like `wrap_tags_in` should make this doable.

Reply:
Worked nicely with the new function, thank you 👍